steering_controllers_library/src/steering_odometry.cpp

```diff
@@ -0,0 +1,341 @@
+/*********************************************************************
+* Copyright (c) 2023, Stogl Robotics Consulting UG (haftungsbeschränkt)
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ *********************************************************************/
+
+/*
+ * Author: dr. sc. Tomislav Petkovic
+ * Author: Dr. Ing. Denis Stogl
+ */
+
+#include "steering_controllers_library/steering_odometry.hpp"
+
+#include <cmath>
+#include <iostream>
+
+namespace steering_odometry
+{
+SteeringOdometry::SteeringOdometry(size_t velocity_rolling_window_size)
```

Comment:
basic unit tests would be nice. This class has 0 coverage on it's own

Reply:
This is intentionally. Since we would need some fake implementation of virtual class which we basically have in the specific implementations. If we test everything, then it's the same as we test implementations itself.